lib/services/user.go

```diff
@@ -22,6 +22,54 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// ValidateUser validates the User and sets default values
+func ValidateUser(u User) error {
```

Comment:
I think it could be less friction to keep these methods on an aliased type? WDYT?
This will definitely help with names such as `LocalAuthSecretsEquals`.

Reply:
I agree, this is something I've been talking about with others and could help differentiate our api types from our business logic / service types. For now the plan is to make the changes like this, but once it hits a critical mass and other significant tasks are complete, I want to refactor this.